src/braket/devices/device.py

```diff
@@ -104,3 +110,43 @@ def status(self) -> str:
             str: The status of this Device
         """
         return self._status
+
+    def _validate_device_noise_model_support(self) -> None:
+        noise_pragma_name_to_class = copy(one_prob_noise_map)
+        noise_pragma_name_to_class.update({"kraus": Noise.Kraus})
+
+        supported_noises = [
+            noise_pragma_name_to_class[pragma.split("braket_noise_")[-1]]
+            for pragma in (self.properties.action[DeviceActionType.OPENQASM].supportedPragmas)
+            if "braket_noise_" in pragma
+        ]
+
+        noise_operators = [noise_instr.noise for noise_instr in self._noise_model._instructions]
+        if not all([isinstance(noise, tuple(supported_noises)) for noise in noise_operators]):
```

Comment:
Nor a big deal, I'll say it anyway: instead of iterating over the supported noises for each noise instruction, how about adding the name of each noise instruction to a set, and then raising the exception if the set difference between this set and the supported noise set is nonempty?

Reply:
Thanks for the suggestion! That would be ideal if a device object has a native attribute of "supported noises". Currently it doesn't. To do it in the way you described, I would need to a way to distinguish noise operations/pragmas from other operations/pragmas. To achieve this, I might need string parsing or add more attributes about noise to the device class.

Reply:
You don't have to use a supported noise attribute; you can build the set directly from the `supported_noises` variable that you already have. Something like this:
```suggestion
        supported_noise_set = set(supported_noise)
        noise_model_operator_set = {noise.name for noise in noise_operators}
        if noise_model_operator_set - supported_noise_set:
```